ucb-rit/coldfront
# Add user access agreement form

Fixes #27

**Changes**
- Added a field `access_agreement_form_date` to `UserProfile` that stores the time at which the user signed the access agreement.
- Added a view at `/user/user-access-agreement`.
- Updated the User Profile to display whether or not it has been signed and, if not, a link to do so.

**How to Test**
- Review the code changes and add comments as needed.
- Ensure that the test suite passes.
- Manual Testing
    - Run `python manage.py migrate`.
    - Navigate to the User Profile. Initially, the entry for the agreement should include a red "X" and a "Sign" button.
    - Sign the user agreement with an incorrect answer for the pop quiz (correct: 24). An error message should appear.
    - Sign the user agreement with a correct answer. A redirect should occur and a success message should appear. The entry in the User Profile should have changed to a green check.
    - Attempt to navigate to the URL for the access agreement. A redirect should occur and a warning message should appear.

**Notes**
- To clear out the field in the database, run the following in `python manage.py shell`:
    ```
    from coldfront.core.user.models import UserProfile
    username = "USERNAME_HERE"
    UserProfile.objects.filter(user__username=username).update(access_agreement_signed_date=None)
    ```
- Currently, the help text for each field in the form appears below the field. This can be corrected at a later date if needed.
